cmake/extensions.cmake

```diff
@@ -1387,14 +1387,7 @@ endfunction()
 
 function(target_ld_options target scope)
   foreach(option ${ARGN})
-    string(MAKE_C_IDENTIFIER check${option} check)
-
-    set(SAVED_CMAKE_REQUIRED_FLAGS ${CMAKE_REQUIRED_FLAGS})
-    set(CMAKE_REQUIRED_FLAGS "${CMAKE_REQUIRED_FLAGS} ${option}")
-    zephyr_check_compiler_flag(C "" ${check})
-    set(CMAKE_REQUIRED_FLAGS ${SAVED_CMAKE_REQUIRED_FLAGS})
-
-    target_link_libraries_ifdef(${check} ${target} ${scope} ${option})
+    target_link_libraries(${target} ${scope} ${option})
```

Comment:
> Some linker options are only valid in combinations, like the RISC-V
> -march -mabi values. Attempting to validate them one-at-a-time will
> fail, even though the combination of values is valid. This could be
> changed to validate the whole set of options provided at one time, which
> would work in the RISC-V case.

We should not remove linker option validation just because it breaks something for RISC-V.

Please consider cherry-picking the following commit into this PR:
https://github.com/zephyrproject-rtos/zephyr/pull/25475/commits/f6a67ecc2f13c7c1efa1c47c32897b1e08e269d5

cc @tejlmand

Reply:
Thanks so much for fixing this in a very clean way. My kludge was definitely not suitable for merging, but did manage to make the picolibc build work. Picolibc ships with a .specs file that computes the correct library name based on the arch/abi values provided to the compiler, so when those weren't passed to the linker, the wrong C library was used causing the link to fail. With your fix, the arguments are correctly passed and linking now works! 

I've cherry-picked that one patch into this PR; I'm happy to wait for your PR to be merged and rebase if you like.

Reply:
Good to hear that it works.

> I've cherry-picked that one patch into this PR; I'm happy to wait for your PR to be merged and rebase if you like.

[That PR](https://github.com/zephyrproject-rtos/zephyr/pull/25475) will take quite some time to get merged. I am okay with that commit being merged as part of this PR. I will make sure to rebase my PR after that happens.

Reply:
Sounds good. I've tested the SDK's arm-zepyr-eabi compiler and it has the necessary TLS support enabled.